### PR TITLE
chore(common): resolve audit issues in Web-related repo NPM packages

### LINKED ISSUE #4612: bug(web): dependency updates needed
**Describe the bug**
Screen Shot 2021-03-05 at 10 12 33 AM

Those aren't good numbers.

Be sure to check on the dependencies for each of the subpackages, too!

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #4612 for early 15.0 alpha.

All changes in this PR are solely to `lerna.json`, `package.json`, and `package-lock.json` files within the repo.  There are no code changes.

Only the following audit issues remain:

Screen Shot 2021-05-17 at 10 20 19 AM

Try as I might, I couldn't get `lerna` to stop stomping on any changes I tried that would fix the `lodash` one.  The other, I left in place b/c the only fix is to move to a package's `beta`.  Of course, both of these are included solely as dependencies of our dev-dependencies, so this isn't the most worrying thing in the world.

Also, to reduce the load of all the NPM stuff, I've set some of the most common dev-dependencies to use hoisting now.
- `mocha` - the common JS test framework (for both Node.js and Karma -> in-browser use)
- `chai` - the common assertion library
- `ts-node` - was required for the lexical model compiler once `mocha` was hoisted.  Allows TS-based unit tests.

All affected subprojects within the repo have had their tests run already on my local (macOS) development machine.  (These - the _tests_ - are the parts of said subprojects that would have been affected.)  Of course, we should also let the Win-based CI tests run their course as well.

------

One thing of note:  you may notice that the package lockfiles' versions have all been updated to version 2.  This is something that automatically happens on a development machine using `npm 7`.  It is noted [to be backward compatible with prior versions of NPM](https://github.com/npm/cli/blob/653769de359b8d24f0d17b8e7e426708f49cadb8/docs/content/configuring-npm/package-lock-json.md#lockfileversion), so there's likely no need to worry:

> Lockfiles generated by npm v7 will contain lockfileVersion: 2.

NPM 5 & 6 use lockfile version 1, so our build agents should not be adversely affected at this time.
> ...
>     2: The lockfile version used by npm v7, which is backwards compatible to v1 lockfiles.

### CODE REVIEW

**Comment by jahorton** [comment]
Ah, looks I need to hoist the TC-reporting package for `mocha` when hoisting `mocha`.

**Review by mcdurdin — COMMENTED** [review]
Looks like there are merge conflicts to resolve before I can approve.

**Comment by jahorton** [comment]
> Looks like there are merge conflicts to resolve before I can approve.

I don't think it wise to resolve them until it _is_ approved.  The changes occur to the package.json files, which are also edited automatically every day if any PRs are merged that day.  So... if there's an overnight delay in the review, it'll just conflict again.

**Comment by jahorton** [comment]
> > Looks like there are merge conflicts to resolve before I can approve.
> 
> I don't think it wise to resolve them until it _is_ approved. The changes occur to the package.json files, which are also edited automatically every day if any PRs are merged that day. So... if there's an overnight delay in the review, it'll just conflict again.

The reason it's showing up like this is likely due in part to NPM 7 and the new lockfile version; it does substantially reorganize things in the package-lock.json files, so I'm not surprised Git's merge stuff doesn't know what to do by default.

File: common/core/web/keyboard-processor/package.json [hunk 1]
```diff
@@ -19,10 +19,11 @@
...
     "@keymanapp/resources-gosh": "^15.0.48",
-    "chai": "^4.2.0",
-    "mocha": "^5.2.0",
-    "mocha-teamcity-reporter": "^3.0.0",
-    "typescript": "^3.8.3"
+    "chai": "^4.3.4",
+    "mocha": "^8.4.0",
+    "mocha-teamcity-reporter": "^4.0.0",
+    "typescript": "^3.8.3",
+    "ts-node": "^9.1.1"
```
Comment (mcdurdin) [inline common/core/web/keyboard-processor/package.json:26]: Good to see `ts-node` move to a dev dependency

**Review by mcdurdin — APPROVED** [review]

**Comment by jahorton** [comment]
Almost missed some top-level stuff; I noticed it during some of my iOS test builds.  For things at the top level, a simple `npm audit fix` works nicely.  It's more tricky when confined within a subfolder.

Screen Shot 2021-05-20 at 8 07 24 AM

Those were fixed by the last, post-merge commit.

File: common/core/web/keyboard-processor/package.json (hunk 1 above)
Comment (jahorton) [reply common/core/web/keyboard-processor/package.json:26]: Yep.  Wasn't as sure what its role was when I first made the package, so it was a pretty simple mistake to correct.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.50-alpha](https://keyman.com/downloads/releases/alpha/15.0.50)

**Pull request merged**